server/pulp/server/webservices/middleware/exception.py

```diff
@@ -31,28 +31,36 @@ def process_exception(self, request, exception):
         :return: Response containing processed exception
         :rtype: django.http.HttpResponseServerError
         """
-        if isinstance(exception, PulpException):
-            status = exception.http_status_code
-            response = error.http_error_obj(status, str(exception))
-            response.update(exception.data_dict())
-            response['error'] = exception.to_dict()
-            logger.info(str(exception))
-            response_obj = HttpResponse(json.dumps(response), status=status,
-                                        content_type="application/json; charset=utf-8")
-        else:
-            status = httplib.INTERNAL_SERVER_ERROR
-            response = error.http_error_obj(status, str(exception))
-            msg = _('Unhandled Exception')
-            logger.error(msg)
-
-        if status == httplib.INTERNAL_SERVER_ERROR:
-            logger.exception(str(exception))
-            e_type, e_value, trace = sys.exc_info()
-            response['exception'] = traceback.format_exception_only(e_type, e_value)
-            response['traceback'] = traceback.format_tb(trace)
-            response_obj = HttpResponseServerError(json.dumps(response),
-                                                   content_type="application/json; charset=utf-8")
-        else:
-            logger.info(str(exception))
-
-        return response_obj
+
+        try:
+            if isinstance(exception, PulpException):
+                status = exception.http_status_code
+                response = error.http_error_obj(status, str(exception))
+                response.update(exception.data_dict())
+                response['error'] = exception.to_dict()
+                logger.info(str(exception))
+                response_obj = HttpResponse(json.dumps(response), status=status,
+                                            content_type="application/json; charset=utf-8")
+            else:
+                status = httplib.INTERNAL_SERVER_ERROR
+                response = error.http_error_obj(status, str(exception))
+                msg = _('Unhandled Exception')
+                logger.error(msg)
+
+            if status == httplib.INTERNAL_SERVER_ERROR:
+                logger.exception(str(exception))
+                e_type, e_value, trace = sys.exc_info()
+                response['exception'] = traceback.format_exception_only(e_type, e_value)
+                response['traceback'] = traceback.format_tb(trace)
+                response_obj = HttpResponseServerError(
+                    json.dumps(response), content_type="application/json; charset=utf-8")
+            else:
+                logger.info(str(exception))
```

Comment:
I know this line was like this before your change, but it seems like this line is going to re-log the str() of the Exception in non INTERNAL_SERVER_ERROR cases. I think we can drop the else and this line and not lose any info.